src/gobworkflow/storage/storage.py

```diff
@@ -345,31 +347,43 @@ def job_get(job_id):
 
 
 @session_auto_reconnect
-def job_runs(jobinfo, msg):
+def job_runs(jobinfo, msg) -> bool:
+    """
+    Checks for job duplicate based on header information, if all equal:
+     - Model:
+         - catalogue
+         - collection
+         - attribute
+         - application
+     - Extra arguments:
+         - destination
+         - source
+         - entity_id
+     - Job age:
+         - less than 12 hours
+
+    Otherwise a job is not a duplicate and should be started.
+    """
     header = msg.get('header')
-
-    # Check job duplicates source/cat/col/attr,
-    # destination is used in export to export to different locations
-    # application is used in import
-    check_args = ['source', 'catalogue', 'collection', 'attribute', 'destination', 'entity_id', 'application']
-    job_args = [header.get(key) for key in check_args if header.get(key)]
-
-    # Filter jobs on type and catalog / collection / attribute if available
-    job = session.query(Job)\
-        .filter(Job.id != jobinfo['id'])\
-        .filter(Job.type == jobinfo['type'])\
-        .filter(cast(Job.args, ARRAY(String)).contains(cast(job_args, ARRAY(String))))\
-        .filter(Job.end == None)\
-        .order_by(Job.start.desc())\
+    check_args = ['catalogue', 'collection', 'attribute', 'application']
+    job_args = [header.get(key) for key in ['destination', 'entity_id', 'source'] if header.get(key)]
+
+    job = (
+        session
+        .query(Job)
+        .filter(Job.id != jobinfo['id'])
+        .filter(Job.type == jobinfo['type'])
+        .filter_by(**{arg: header.get(arg) for arg in check_args})
+        .filter(cast(Job.args, ARRAY(String)).contains(cast(job_args, ARRAY(String))))
+        .filter(Job.end == None)
+        .order_by(Job.start.desc())
         .first()  # noqa E711 (== None)
-    if job is None:
-        return False
-    else:
-        # Consider jobs of less than 12 hours old as still running
-        duration = datetime.datetime.now() - job.start
-        zombie = duration.total_seconds() >= (12 * 60 * 60)
-        print("Found already running job", job.id, job.start, duration, zombie)
-        return not zombie
+    )
+    if has_job_run := job is not None:
```

Comment:
i dont think the walrus operator makes this easier to read tbh


```suggestion
      if job is None:
          return False
        
      print(f"Found already running job '{job.id}', started {job.start} (zombie: {job.is_zombie()})")
      return not job.is_zombie()
          
```